### PR TITLE
util-timer merge

### PR DESCRIPTION
**Pull request by Jurgenvs**
Merging timer with dev

### CODE REVIEW

File: core/src/cg/group4/view/MainMenu.java [hunk 1]
```diff
@@ -1,105 +1,152 @@
...
-               System.out.println("Settings");
-            }
-        });
-
-        stage.addActor(button);
-	}
-
-	@Override
-	public void render(float delta) {
-		Gdx.gl.glClearColor(0, 132 / 255f, 197 / 255f, 1);
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/MainMenu.java]: No magic numbers

File: core/src/cg/group4/view/MainMenu.java [hunk 2]
```diff
@@ -1,105 +1,152 @@
...
+        };
+        TimeKeeper.getInstance().getTimer("INTERVAL").subscribe(timerTask);
+    }
+
+    /**
+     * @param delta Delta time
+     */
+    @Override
+    public final void render(float delta) {
+        Gdx.gl.glClearColor(0, 132 / 255f, 197 / 255f, 1);
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/MainMenu.java]: Still no magic numbers...

**Comment by Surskit** [comment]
c_attribute moet overal worden verandert in cAttribute.

File: core/test/cg/group4/LauncherTest.java [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
-		int x = 5;
-		assertTrue(x == 5);
-	}
-	
-	@Test
-	public void test1() {
-		assertTrue(Launcher.testerooni() == 2);
-	}
+    @Test
+    public void test() {
```
Comment (sirpolomotro) [inline core/test/cg/group4/LauncherTest.java]: USELESS TEST!!!

File: core/test/cg/group4/util/timer/TimeKeeperTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+
+import java.util.HashSet;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(GdxTestRunner.class)
```
Comment (sirpolomotro) [inline core/test/cg/group4/util/timer/TimeKeeperTest.java]: Not full coverage

File: core/src/cg/group4/view/Settings.java [hunk 5]
```diff
@@ -1,47 +1,111 @@
...
+		cButtonResetStroll.addListener(new ChangeListener() {
+            @Override
+            public void changed(final ChangeEvent event, final Actor actor) {
+                TimeKeeper.getInstance().getTimer("STROLL").reset();
+            }
+        });
+        cButtonBack.addListener(new ChangeListener() {
+            @Override
+            public void changed(final ChangeEvent event, final Actor actor) {
+                ((Game) Gdx.app.getApplicationListener()).setScreen(new MainMenu());
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/Settings.java]: Main menu already exists, should not be new instance

File: core/src/cg/group4/view/MainMenu.java [hunk 6]
```diff
@@ -1,105 +1,152 @@
...
+
+        TimerTask timerTask = new TimerTask() {
+            @Override
+            public void onTick(int seconds) {
+                cButtonStroll.setText("Stroll: " + seconds);
+            }
+
+            @Override
+            public void onStart() {
+                cButtonStroll.setText("Stroll: ");
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/MainMenu.java]: Looks ugly, get delay..?

File: core/src/cg/group4/util/timer/TimeKeeper.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    /**
+     * Previous tick that the timers were called
+     */
+    protected long cPreviousTick;
+
+    /**
+     * Amount
+     */
+    protected final long cMillisInSecond = 1000;
```
Comment (sirpolomotro) [inline core/src/cg/group4/util/timer/TimeKeeper.java]: Not consistent with same variable in Timer

**Pull request merged**